docs/Ways-of-Working/Toolkit/Test-Engineering/Best-Practices/Playwright-Test-Automation-Guidelines.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Playwright Test Automation Guidelines
+authors: 
+- Jees Mathew 
+- Karen Farrell 
+- Mihajlo Stojanovski 
+- James Perry
+tags:
+    - Default
+    - Test
+---
+
+## Introduction
+
+This document outlines best practices to follow when writing Playwright test scripts.
+
+## Locating selectors
+
+When planning testing at the readying phase of a work item, locators must be considered.  This allows Software Engineers to plan any additional work needed in order to enable our best practices.
+
+It is important to use a locator that is robust and does not depend on the DOM structure.
+
+To make our tests resilient, we prioritise user-facing attributes and explicit contracts when locating elements on a page.  This could be text, a button name, an element's aria-label etc.
+
+Playwright recommended built-in locators:
+
+- page.getByRole() to locate by explicit and implicit accessibility attributes.
+- page.getByText() to locate by text content.
+- page.getByLabel() to locate a form control by associated label's text.
+- page.getByPlaceholder() to locate an input by placeholder.
+- page.getByAltText() to locate an element, usually image, by its text alternative.
+- page.getByTitle() to locate an element by its title attribute.
+
+
+If an element cannot be located by any of these, then we look for data-test IDs (for more info click [here](https://playwright.dev../../locators#locate-by-test-id)).  **Software Engineers will support creating the data-test-IDs where needed.**
```

Comment:
Some Test Engineers can create these too so think we should update this text

Reply:
updated :)